api/src/main/java/org/openmrs/module/coreapps/customdatatype/CodedConceptDatatype.java

```diff
@@ -0,0 +1,29 @@
+package org.openmrs.module.coreapps.customdatatype;
+
+import org.openmrs.customdatatype.SerializingCustomDatatype;
+import org.springframework.stereotype.Component;
+
+/**
+ * This class allows for coded concepts to be displayed as a drop-down when creating visit attributes.
+ * The concept id MUST be set in the datatype configuration field.
+ */
+@Component
+public class CodedConceptDatatype extends SerializingCustomDatatype<String> {
+
+	/**
+	 * @see org.openmrs.customdatatype.SerializingCustomDatatype#serialize(java.lang.Object)
+	 */
+	@Override
+	public String serialize(String typedValue) {
+		return typedValue;
+	}
+
+	/**
+	 * @see org.openmrs.customdatatype.SerializingCustomDatatype#deserialize(java.lang.String)
+	 */
+	@Override
+	public String deserialize(String serializedValue) {
```

Comment:
Ok. Notice this commit is on branch 1.7 which didn't yet have ConceptDatatype (introduced in 2.0)